base/src/commonMain/kotlin/monitor/ServiceMonitor.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ Copyright 2021 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.base
+
+import com.splendo.kaluga.logging.debug
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+
+abstract class ServiceMonitor {
+
+    abstract val TAG: String
```

Comment:
You can do
```
val TAG = this::class.simpleName!!
```

Reply:
it will be "ServiceMonitor", not Bluetooth or Location

Reply:
```abstract class A {
    val x: String = this::class.simpleName!!
}

class B : A()

fun main() = runBlocking {
    val a = B().x
    println("class is $a")
}
```

I see `class is B`

Reply:
b093d8f